docs/advanced-topics/client-anticipation.md

```diff
@@ -0,0 +1,117 @@
+
+### Global OnReanticipate
+
+In addition to the `NetworkBehaviour`'s `OnReanticipate` method, `NetworkManager` also has a callback that can be subscribed to for global reanticipation. This is useful if you need to run your reanticipation in a more global way, such as if you need to run it step-wise (say, anticipating one frame at a time) and need all objects to complete one step before any of them begin the second one. This callback receives the same `lastRoundTripTime` value as the `NetworkBehaviour` method, and is called **after** all of the `NetworkBehaviour` methods have been called.
+
+:::note
+
+If you want to implement a full client-side prediction model in your game, the global OnReanticipate callback is likely the ideal place to incorporate your rollback and replay logic. The details of implementing this, however, are left as an exercise for the user; implementing a full, production-ready prediction loop is a complex topic and recommended for advanced users only.
+
+:::
+
+## Smoothing
+
+Much of the time, when you perform an anticipation action, your anticipation will be correct and the server will update the state to the same value as you did. Occasionally, though, the server will have a different result, since latency can result in the server state changing before it receives your command in ways that affect the result. When this happens, by default, the value will be snapped to the new server value.
+
+In some cases, this may result in an undesirable player experience, causing things that normally move smoothly to become choppy. When this happens, you can smooth out the result by using the `Smooth()` function on both `AnticipatedNetworkVariable` and `AnticipatedNetworkTransform`.
+
+`Smooth()` takes a starting position and an ending position (which would usually be the prevoius anticipated value and either the new anticipated value or the current authoritatvie value) and a duration over which to perform the smoothing. For `AnticipatedNetworkVariable`, because the type is generic, it also requires a delegate to use to perform the smoothing operation (which takes a from value, a to value, and a percent value, and in many cases may just be `Mathf.Lerp`).
+
+Because the first parameter to `Smooth()` is usually the previous anticipated value, and by the time the `OnReanticipate` function is called, the anticipated value has been overwritten with the authoritative value, you can use `PreviousAnticipatedValue` on `AnticipatedNetworkVariable` and `PreviousAnticipatedState` on `AnticipatedNetworkTransform` to access the value you most recently passed into `Anticipate()` or its related transform methods. Note, however, that if you call `Anticipate()` or its related transform methods prior to doing the smoothing, they will update the previous anticipated value, so you may need to store the previous anticipated value in a local variable before doing your reanticipation.
```

Comment:
Hopefully that's something Amy can help with, that doc has very long sentences :P